notify/impl.go

```diff
@@ -198,7 +198,7 @@ func (w *Webhook) Notify(ctx context.Context, alerts ...*types.Alert) error {
 	resp.Body.Close()
 
 	if resp.StatusCode/100 != 2 {
-		return fmt.Errorf("unexpected status code %v", resp.StatusCode)
+		return fmt.Errorf("Webhook: unexpected status code %v from %s", resp.StatusCode, w.URL)
```

Comment:
OK, I tried figuring out how this would work when the logging is made from the RetryNotifier - which has a reference to a Notifier instance, but not an Integration. Now, as I'm a complete Go noob I didn't find a trivial and non-ugly way to do this based on what I had in https://github.com/prometheus/alertmanager/blob/master/notify/notify.go#L193 - I tried a few things like modifying the Notifier interface to also include a name and such, but ... well, I'll just say I wasn't satisfied with what I came up with, so I never got around to responding to this.

I can remove these integration names from the error strings entirely if you like? They were very handy for pinning down the problem that led me to submitting this PR, though. =)